### PR TITLE
Update checkbox

### LINKED ISSUE #400: Update checkbox
https://material-ui.com/components/checkboxes/

### PR DESCRIPTION
**Pull request by dwkaminsky**
## Description
Updates Checkbox to the latest `material-ui`

## Motivation and Context
Closes #400 

## Screenshots (if appropriate):
captured (3)

### CODE REVIEW

File: src/lib/components/Checkbox.react.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import getMuiTheme from 'material-ui/styles/getMuiTheme';
+
+type Props = {
+  /** Checkbox is checked if true */
+  checked?: boolean,
+  /** CSS class name of the root element */
+  className?: string,
+  /** Mapping from MUI Element name to the intended classname for that Element
+   * See https://material-ui.com/api/checkbox/#css*/
+  classes?: Object,
```
Comment (coralvanda) [inline src/lib/components/Checkbox.react.js:17]: Please add full set of options with type hinting.

File: usage.py [hunk 2]
```diff
@@ -145,6 +197,31 @@ def show_modal_dialog(modal_click: int, close_button: int, open_state: bool):
...
+                        _check_1: bool, _check_2: bool, _check_3: bool,
+                        name_1: str, name_2: str, name_3: str, ):
+    output = []
+    if _check_1:
+        output.append(name_1)
+    if _check_2:
+        output.append(name_2)
+    if _check_3:
+        output.append(name_3)
+    return f'{output}'
```
Comment (coralvanda) [inline usage.py:222]: Does this not work correctly unless you use the `State` of the `checked` prop?

File: src/lib/components/Checkbox.react.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+export default class Checkbox extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {checked: props.checked};
+  }
+
+  UNSAFE_componentWillReceiveProps(nextProps: Props, nextContext: *): void {
+    if (nextProps.checked !== null && nextProps.checked !== this.props.checked) {
+      this.props.setProps({checked: nextProps.checked})
+      // this.setState({checked: nextProps.checked});
```
Comment (coralvanda) [inline src/lib/components/Checkbox.react.js:60]: Will the checkbox update correctly from another component (say a button to clear all checks) without this `setState` call?

Comment (dwkaminsky) [reply src/lib/components/Checkbox.react.js]: Great catch

File: usage.py [hunk 4]
```diff
@@ -203,24 +179,32 @@ def show_modal_dialog(modal_click: int, close_button: int, open_state: bool):
...
 
+@app.callback(
+    [dash.dependencies.Output('checkbox1', 'checked'),
+     dash.dependencies.Output('checkbox2', 'checked'),
+     dash.dependencies.Output('checkbox3', 'checked')],
+    [dash.dependencies.Input('clear-checks', 'n_clicks')])
+def callback_checkboxes(n: int):
+    if not n:
+        raise dash.exceptions.PreventUpdate
+    return False, False, False
```
Comment (coralvanda) [inline usage.py:204]: This is good, thank you for adding this example!

**Pull request merged**